doc/test-framework/scorecard.md

```diff
@@ -79,29 +84,245 @@ API server, indicating that it is modifying resources. This test has a maximum s
+        },
+        "partialPass": {
+          "type": "integer",
+          "description": "PartialPass is the number of tests that ended in the PartialPass state"
+        },
+        "fail": {
+          "type": "integer",
+          "description": "Fail is the number of tests that ended in the Fail state"
+        },
+        "totalTests": {
+          "type": "integer",
+          "description": "TotalTests is the total number of tests run in this suite"
+        },
+        "totalScorePercent": {
+          "type": "integer",
+          "description": "TotalScorePercent is the total score of this suite as a percentage"
+        },
+        "tests": {
+          "items": {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "$ref": "#/definitions/ScorecardTestResult"
+          },
+          "type": "array"
+        },
+        "log": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "ScorecardTestResult": {
+      "required": [
+        "state",
+        "name",
+        "description",
+        "earnedPoints",
+        "maximumPoints",
+        "suggestions",
+        "errors"
+      ],
+      "properties": {
+        "state": {
+          "type": "string",
+          "description": "State is the final state of the test. Valid values are: pass, partial_pass, fail, error"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name is the name of the test"
+        },
+        "description": {
+          "type": "string",
+          "description": "Description describes what the test does"
+        },
+        "earnedPoints": {
+          "type": "integer",
+          "description": "EarnedPoints is how many points the test received after running"
+        },
+        "maximumPoints": {
+          "type": "integer",
+          "description": "MaximumPoints is the maximum number of points possible for the test"
+        },
+        "suggestions": {
+          "items": {
+            "type": "string"
+          },
+          "type": "array",
+          "description": "Suggestions is a list of suggestions for the user to improve their score (if applicable)"
+        },
+        "errors": {
+          "items": {
+            "type": "string"
+          },
+          "type": "array",
+          "description": "Errors is a list of the errors that occured during the test (this can include both fatal and non-fatal errors)"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    }
+  }
+}
+```
+
+**NOTE:** The `ScorecardOutput` object is designed the same as a Kubernetes API, and thus also has a full `TypeMeta` and `ObjectMeta`. Those are optional and not
+used by the scorecard at the current time. These optional fields have thus been removed from the schema for cleanliness and the schema only includes the fields
+the scorecard currently looks at.
```

Comment:
Do you think we should make these required now and start checking for them? The other option is to allow them to be optional for `v1alpha1` and required for all future revisions. I'm okay with either option, but if we go with the latter options, I think we should add another sentence to this note that says these fields will be required in future revisions to this API.

Reply:
While we don't actually check `apiVersion` yet (which we should), I meant that field to be required as well as the `kind` (which is why those are marked as required in the JSONSchema). I was referring to other fields in `ObjectMeta`, such as `selfLink`, `uid`, `creationTimestamp`, etc. are not required at the current time. I'll rephrase this to make that clear.

Reply:
Updated note